### PR TITLE
Fix 500 when using system profile filter on integer fields

### PR DESCRIPTION
**Pull request by ryandillinfelton**
# Overview

This PR is being created to address [ESSNTL-2039](https://issues.redhat.com/browse/ESSNTL-2039).

The bug was caused by a lack of support for integer fields in system profile. According to past Ryan in a comment xjoin didn't support ints at the time.

I've simply added in a new filter constructor for integers and updated a few enums. I also added tests for integer type fields created by generic filtering in the style of the existing ones for strings, booleans, etc.

This doesn't expand the field to allow for range operations [gt, gte, lt, lte]. I've created a [follow up task](https://issues.redhat.com/browse/ESSNTL-2084) to add that in since there are no existing generic filtering types that support range operations and the implementation could be more complex than I expect. I'd rather keep this fix small and concise.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [x] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [x] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: tests/test_xjoin.py [hunk 1]
```diff
@@ -2922,6 +2922,156 @@ def test_generic_filtering_booleans_invalid_values(subtests, query_source_xjoin,
...
+                        url = url_builder(query=f"?filter{path}{op}={value}")
+
+                        response_status, _ = api_get(url)
+                        assert response_status == 200
+
+                        query_verifier(mocker, query_mock, query)
+                        query_mock.reset_mock()
+
+
+# having both system_profile endpoints creates a moching issue right now.
```
Comment (kruai) [inline tests/test_xjoin.py:2986]: Just a lil copy&paste fix
```suggestion
# having both system_profile endpoints creates a mocking issue right now.
```

**Review by kruai — APPROVED** [review]
Very thorough testing. Minus one typo (leftover from copy&paste), this all looks good to me

**Review by roliveri — DISMISSED** [review]

File: tests/test_xjoin.py (hunk 1 above)
Comment (ryandillinfelton) [reply tests/test_xjoin.py:2986]: Haha, I can't believe I fixed that in the original test but not I'm my copypaste of it 🤦

**Review by kruai — APPROVED** [review]

**Review by thearifismail — APPROVED** [review]
LGTM

**Pull request merged**